source/main/utils/launcherConfig.js

```diff
@@ -0,0 +1,27 @@
+import { readFileSync } from 'fs';
+
+const yamljs = require('yamljs');
+
+function getLauncherConfig() {
+
+  if (process.env.LAUNCHER_CONFIG) {
+    console.log('Getting Data Directory from Launcher config');
+    const inputYaml = readFileSync(process.env.LAUNCHER_CONFIG, 'utf8');
+    if (process.env.XDG_DATA_HOME === undefined) {
+      process.env.XDG_DATA_HOME = process.env.HOME + '/.local/share/';
```

Comment:
Constructing paths like this could be problematic on Windows, right?

And why is this variable set here? Seems like an odd place to set an environment flag?
Additionally this is confusing because it is named like a simple getter function but then it has this (probably important?) side effect.

Reply:
linux usage refers to `$XDG_DATA_HOME` which has a default value when not set
windows never refers to that variable, so this path wont be used on windows

Reply:
Ok thanks for the info … still i think it's the wrong place to set an env variable. And at least it should be commented that this is only relevant for linux etc. … (basically what you just wrote would have helped tremendously here to understand whats going on)

Reply:
@DominikGuzei @cleverca22 I have now added comments to this part of the code